### PR TITLE
leaseoptions: add namespace to LeaseOptions to return on responses

### PR DESCRIPTION
**Pull request by calvn**

### CODE REVIEW

**Comment by calvn** [comment]
The generated proto is probably not right. Using the latest proto plugin will update `ProtoPackageIsVersion2` to `ProtoPackageIsVersion3` which we might want to do in a separate PR.

File: logical/translate_response.go [hunk 1]
```diff
@@ -92,6 +94,7 @@ type HTTPResponse struct {
 	Warnings      []string               `json:"warnings"`
 	Headers       map[string][]string    `json:"-"`
 	Auth          *HTTPAuth              `json:"auth"`
+	Namespace     string                 `json:"namespace,omitempty"`
```
Comment (briankassouf) [inline logical/translate_response.go]: I don't think we need to add this to the top level response object here

Comment (calvn) [reply logical/translate_response.go]: The fields in `logical.Response.Secret` gets translated into the top-level field of `HTTPResponse`, see https://github.com/hashicorp/vault/pull/6212/files#diff-149879c600c98fcad95ce83cf71dea10R25 for `Response.Secret.Namespace` as well as other Secret related values.

**Comment by jefferai** [comment]
There's a problem here which is that namespace in and of itself isn't a meaningful thing. There's namespace path, which could be removed and readded, and namespace ID, which won't. If there is only one value to be used I think it has to be namespace ID or you could run into errors trying to renew a value produced in a previous version of the namespace.

Brian said the other day that we do append nsid to lease values, which makes me wonder why this is necessary in the first place.

**Comment by calvn** [comment]
The issue that this PR was trying to address is renewal of leases on namespaces in the case where the namespace is provided as the full path. On the client's side, in this case the api.Renewer's client, there's no way to determine the namespace value to set as its header on based the nsid (or on the leaseID, since it wouldn't really know where to split the string). We would be hitting this error in this case: https://github.com/hashicorp/vault/blob/master/vault/expiration.go#L834

**Comment by jefferai** [comment]
If the nsid is on lease values can't we split it the same way we split tokens?

**Comment by calvn** [comment]
Yes, but from the client's perspective, there's no way to derive the namespace path to use from that nsid. For example, let's say that the lease `kv/foo/H80orMpo3Vr9MZWmXVaT6rL2.LmlV3` belongs to ns1. The _client_ can't really determine what's the namespace for that lease based off that string so it can't construct the request proper renewal request (either by setting `ns1/` as the namespace header value or by constructing `ns1/sys/leases/renew` as the renewal path).

**Comment by jefferai** [comment]
Is there a use case for renewing a lease in a different namespace than it originated from?

**Comment by calvn** [comment]
I don't think we allow that. We currently require the renewal request to be made on the same namespace that the lease belongs to.

**Comment by jefferai** [comment]
IIRC for tokens we force the request into the right namespace for renewals, couldn't we just do that here too? Then no client ever has to worry about setting the namespace for a renewal, it happens automatically.

**Comment by briankassouf** [comment]
Closing in favor of adding server side handling of lease namespaces

**Pull request closed without merging**